### ISSUE TITLE
Auto-generated test failure

### ISSUE THREAD

**Issue by nivkner** [rust-lang-nursery/rust-bindgen#1318]
This is the smallest case I got, based on clangs internal definition of `max_align_t`, in the header `__stddef_max_align_t.h` used in the libc header `stddef`:

```
typedef struct {
  long double __clang_max_align_nonce2
      __attribute__((__aligned__(__alignof__(long double
))));
} max_align_t;
```
Produces the following error:
```

thread 'header::bindgen_test_layout_max_align_t'
 panicked at 'assertion failed: `(left == right)`
  left: `8`,
 right: `16`: Size of: max_align_t', src/header.rs:10:5

```

system info:
```
rustc 1.27.0-nightly                                   
 binary: rustc                                         
  commit-hash: unknown
commit-date: unknown                                    
host: aarch64-linux-android                         
release: 1.27.0-nightly                             
LLVM version: 6.0



libclang: 6.0.0-2
bindgen 0.37.0
```
The commit is unknown because I'm using the rustc-nightly package for termux from the "its-pointless" repository.

**Comment by emilio** [comment]
Does it happen if you enable repr(align) with rust_target? Otherwise it's expected, we can't just represent 16 bytes of alignment.

I guess now that i128 / u128 are stable we could, but we have repr align too...

**Comment by nivkner** [comment]
How do I enable repr(align)? rust_target only has rustc versions and setting it to nightly doesn't change anything.

**Comment by emilio** [comment]
Setting it to nightly, assuming nightly is new enough, should enable repr(align) support.

**Comment by emilio** [comment]
I assuming this worked? Otherwise I don't want to lose track of it

**Comment by nivkner** [comment]
Sorry, I can't verify this right now, I just thought this won't be fixed until repr(align) is stable, so didn't want to keep an unactionable issue laying around. Keeping it open if it helps.

**Comment by nivkner** [comment]
I inserted repr(align(16)) manually into the generated code and the tests passed. It didn't get auto generated when I used `--rust-target nightly`.

**Comment by nivkner** [comment]
I check my nightly version: it does support repr(align(x)), 
built bindgen from git and used `--rust-target nightly`.
Still generates repr(C) for this code.

**Comment by emilio** [comment]
Yeah, I could repro this... `long double` is... fun, and I don't think it has a way we can represent it in rust other than `f64`...

**Comment by emilio** [comment]
This should be fixed now after #1391 and related work, as long as you use a rust target that supports repr(align).
